eth/catalyst/simulated_beacon.go

```diff
@@ -41,36 +43,47 @@ const devEpochLength = 32
 // withdrawalQueue implements a FIFO queue which holds withdrawals that are
 // pending inclusion.
 type withdrawalQueue struct {
-	pending chan *types.Withdrawal
+	pending types.Withdrawals
+	mu      sync.Mutex
+	feed    event.Feed
+	subs    event.SubscriptionScope
 }
 
+type newWithdrawalsEvent struct{ Withdrawals types.Withdrawals }
+
 // add queues a withdrawal for future inclusion.
 func (w *withdrawalQueue) add(withdrawal *types.Withdrawal) error {
-	select {
-	case w.pending <- withdrawal:
-		break
-	default:
-		return errors.New("withdrawal queue full")
-	}
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	w.pending = append(w.pending, withdrawal)
+	w.feed.Send(newWithdrawalsEvent{types.Withdrawals{withdrawal}})
```

Comment:
Is it a requirement that the `Send` happens inside the mutex? Otherwise that would typically be good to not have inside the mutex-area

Reply:
Ah you're right, no need for that to be in mutex.